### PR TITLE
Paging for list component

### PR DESCRIPTION
**Pull request by vadimbasko**

### CODE REVIEW

File: packages/cuba-react-ui/src/ui/paging/Paging.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+/**
+ * Paging defaults
+ */
+export const defaultPagingConfig: PaginationConfig = {
+  current: 1,
+  pageSize: 10,
+  defaultPageSize: 10,
+  showSizeChanger: true,
+  pageSizeOptions: ['10', '20', '30']
```
Comment (web-devel) [inline packages/cuba-react-ui/src/ui/paging/Paging.tsx]: Let's stick with page sizes like in generic UI 20,50,100,500,1000,5000

File: packages/cuba-react-ui/src/ui/paging/Paging.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * Ant Design Pagination component adopted to our List and DataCollectionStore
+ */
+export class Paging extends React.Component<Props> {
+
+  render() {
+    const {paginationConfig, total} = this.props;
+
+    return (
+      <Pagination
+        {...paginationConfig}
```
Comment (web-devel) [inline packages/cuba-react-ui/src/ui/paging/Paging.tsx]: I wonder if we should introduce our component at all

File: packages/cuba-react-ui/src/ui/paging/Paging.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+ * Returns new PaginationConfig object with 'current' and 'pageSize' params, parsed from location.search.
+ * If param not found in search url, previous value (currentPrev, pageSizePrev) will be returned.
+ *
+ * @param locationSearch trying to find paging params here
+ * @param currentPrev - previous value, will be returned in result if new value not found in locationSeach
+ * @param pageSizePrev - previous value, will be returned in result if new value not found in locationSeach
+ */
+export function parsePagingParams(locationSearch: string, currentPrev: number | undefined, pageSizePrev: number | undefined): PaginationConfig {
+  // remove '?' from start of location.search
+  const locationSearchParams = locationSearch && locationSearch.length > 0 ? locationSearch.substr(1) : '';
```
Comment (web-devel) [inline packages/cuba-react-ui/src/ui/paging/Paging.tsx]: do we need to remove it manually?

from query string docs:

```
console.log(location.search);
//=> '?foo=bar'
 
const parsed = queryString.parse(location.search);
console.log(parsed);
//=> {foo: 'bar'}
```

File: packages/front-generator/src/generators/react-typescript/entity-management/template/Cards.tsx.ejs [hunk 4]
```diff
@@ -19,7 +19,7 @@ class <%= listComponentName %>Component extends React.Component<MainStoreInjecte
   dataCollection = collection<<%= entity.className %>>(<%= entity.className %>.NAME, {view: '<%= listView.name %>', sort: '-updateTs'});
 
   fields = [
-  <% listAttributes.forEach(p => { %>
+  <% listAttributes.forEach(p => { -%>
```
[the remark this answers is not in the document]
Comment (vadimbasko) [reply packages/front-generator/src/generators/react-typescript/entity-management/template/Cards.tsx.ejs]: Yes. It allows to not add superfluous blank line between code lines.

File: packages/cuba-react-ui/src/ui/paging/Paging.tsx (hunk 2 above)
Comment (vadimbasko) [reply packages/cuba-react-ui/src/ui/paging/Paging.tsx]: I vote for introduce component. It's help us to support generated code (and ejs templates) more compact and readable. Also we get rid of copy paste in each generated list.

File: packages/cuba-react-ui/src/ui/paging/Paging.tsx (hunk 3 above)
Comment (vadimbasko) [reply packages/cuba-react-ui/src/ui/paging/Paging.tsx]: Can't find anything about question mark
https://nodejs.org/api/querystring.html

File: packages/cuba-react-ui/src/ui/paging/Paging.tsx (hunk 1 above)
Comment (vadimbasko) [reply packages/cuba-react-ui/src/ui/paging/Paging.tsx]: Done.

File: packages/front-generator/src/generators/react-typescript/entity-management/template/List.tsx.ejs [hunk 5]
```diff
@@ -83,7 +100,15 @@ class <%= listComponentName %>Component extends React.Component<MainStoreInjecte
                   </div>
                 </List.Item>
             }/>
-      </div>);
+
+        {!this.props.paginationConfig.disabled && (
+          <div style={{margin: "12px 0 12px 0", float: "right"}}>
+            <Paging {... this.props} total={this.dataCollection.count} />
```
[the remark this answers is not in the document]
Comment (vadimbasko) [reply packages/front-generator/src/generators/react-typescript/entity-management/template/List.tsx.ejs]: We need also on paging change
```
          <Paging paginationConfig={this.props.paginationConfig}
                  onPagingChange={this.props.onPagingChange}
                  total={this.dataCollection.count}/>
```

File: packages/front-generator/src/generators/react-typescript/entity-management/template/List.tsx.ejs [hunk 6]
```diff
@@ -1,30 +1,48 @@
...
 
   fields = [
-  <% listAttributes.forEach(p => { %>
+  <% listAttributes.forEach(p => { -%>
     '<%= p.name %>',
   <% }) %>
   ];
 
+  componentDidMount(): void {
+    reaction(() => this.props.paginationConfig,
```
[the remark this answers is not in the document]
Comment (vadimbasko) [reply packages/front-generator/src/generators/react-typescript/entity-management/template/List.tsx.ejs]: Fixed.

**Pull request merged**